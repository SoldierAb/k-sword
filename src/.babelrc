{
    "presets": [
      [
        "@babel/preset-env",
        {
          "modules":false,
          "useBuiltIns": "usage",
          "corejs": {
            "version": 3,
            "proposals": true // 使用尚在“提议”阶段特性的 polyfill
          }
        }
      ],
      "@babel/preset-typescript"
    ],
    "plugins": [
      [
        "@babel/plugin-proposal-decorators",
        {
          "legacy": true
        }
      ],
      "@babel/proposal-class-properties"
    ]
}
