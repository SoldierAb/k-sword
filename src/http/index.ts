import axios, { AxiosRequestConfig, AxiosResponse, AxiosInstance } from 'axios'
import { bind } from '../common/bind'
import { extend } from '../common/extend'

export interface HttpInstance extends AxiosInstance {
  [key: string]: any
}

export class Ajax {
  [key:string]:any

  public _config: AxiosRequestConfig

  __CREATED__ = false

   $: HttpInstance

  public requestConfig: (config: AxiosRequestConfig) => AxiosRequestConfig | Promise<AxiosRequestConfig>

  public responseRes: (res: AxiosResponse) => AxiosResponse | Promise<AxiosResponse>

  public requestError: (error: any) => any

  public responseError: (error: any) => any

  constructor(opt: AxiosRequestConfig = {
    headers: {
      "Content-Type": "application/json;charset=UTF-8"
    },
    method:'GET',
    timeout: 5000,
  }) {
    this._config = opt
    this.$ = axios.create(this._config)
    this.requestConfig = opt => (opt.baseURL = '/', opt)
    this.requestError = err => Promise.reject(err)
    this.responseRes = res => res
    this.responseError = err => Promise.reject(err)
    this.setInterceptors()
    this.__CREATED__ = true
  }

  setInterceptors():void {
    this.$.interceptors.request.use(conf => {
      return this.requestConfig(conf)
    }, err => {
      return this.requestError(err)
    })
    this.$.interceptors.response.use(res => {
      return this.responseRes(res)
    }, err => {
      return this.responseError(err)
    })
  }
}

export function createAjax():Ajax{
  const ajax = new Ajax()
  const instance =  bind(ajax.$.request,ajax)
  const _a = extend(instance, ajax)
  for (const key in ajax) {
    Object.defineProperty(_a, key, {
      get() {
        return ajax[key]
      },
      set(newVal) {
        ajax[key] = newVal;
      }
    })
  }
  for (const key in ajax.$) {
    Object.defineProperty(_a, key, {
      get() {
        return ajax.$[key]
      },
      set(newVal) {
        ajax.$[key] = newVal;
      }
    })
  }
  return _a
}

export const ajax = createAjax()
