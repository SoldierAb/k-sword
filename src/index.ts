export { ajax } from './http/index'
export { conf } from './common/config'
export { cookies } from './common/cookies'
export { debounce } from './common/debounce'
export { EventBus } from './common/EventBus'
export { flat } from './common/flat'
export { formatNumber } from './common/formatNumber'
export { getBroswer } from './common/getBroswer'
export { env } from './common/getEnvironment'
export { isArray } from './common/isArray'
export { isBoolean } from './common/isBoolean'
export { isChineseIDCardNumber } from './common/isChineseIDCardNumber'
export { isDate } from './common/isDate'
export { isFunction } from './common/isFunction'
export { isNull } from './common/isNull'
export { isNumber } from './common/isNumber'
export { isNumeric } from './common/isNumeric'
export { isObject } from './common/isObject'
export { isPossibleChineseMobilePhoneNumber } from './common/isPossibleChineseMobilePhoneNumber'
export { isPromise } from './common/isPromise'
export { isString } from './common/isString'
export { isSymbol } from './common/isSymbol'
export { isUndefined } from './common/isUndefined'
export { isUrl } from './common/isUrl'
export { md5 } from './common/md5'
export { readFile } from './common/readFile'
export { throttle } from './common/throttle'
export { tpl } from './common/tpl'
export { type } from './common/type'
export { parallel } from './common/parallel'
export { union } from './common/union'
export { upload } from './common/upload'
export { wait } from './common/wait'
export { loadScript } from './common/loadScript'
export { randomHash } from './common/randomHash'
