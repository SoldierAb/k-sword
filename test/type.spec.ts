import {type} from '../src'


describe('type unit',()=>{
  it('get Undefined',()=>{
    expect(type()).toMatchSnapshot()
  })
  it('get Number',()=>{
    expect(type(1)).toMatchSnapshot()
  })
  it('get String',()=>{
    expect(type(`dasdasd`)).toMatchSnapshot()
  })
  it('get Function',()=>{
    expect(type(new Function())).toMatchSnapshot()
  })
  it('get Array',()=>{
    expect(type([])).toMatchSnapshot()
  })
  it('get Symbol',()=>{
    expect(type(Symbol(1))).toMatchSnapshot()
  })
})
