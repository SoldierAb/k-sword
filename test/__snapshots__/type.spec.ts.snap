// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type unit get Array 1`] = `"Array"`;

exports[`type unit get Function 1`] = `"Function"`;

exports[`type unit get Number 1`] = `"Number"`;

exports[`type unit get String 1`] = `"String"`;

exports[`type unit get Symbol 1`] = `"Symbol"`;

exports[`type unit get Undefined 1`] = `"Undefined"`;
